### PR TITLE
[C-Api] function to check hw availability

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add new function to check hw availability with custom option string.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: c/include/nnstreamer.h [hunk 1]
```diff
@@ -1263,6 +1263,24 @@ int ml_tensors_data_set_tensor_data (ml_tensors_data_h data, unsigned int index,
  */
 int ml_check_nnfw_availability (ml_nnfw_type_e nnfw, ml_nnfw_hw_e hw, bool *available);
 
+/**
+ * @brief Checks the availability of the given execution environments.
```
Comment (gichan-jang) [inline c/include/nnstreamer.h:1267]: It's the same as L1251.
```suggestion
 * @brief Checks the availability of the given execution environments with custom option.
```

Comment (jaeyun-jung) [reply c/include/nnstreamer.h]: updated

**Review by gichan-jang — APPROVED** [review]
LGTM

File: c/src/nnstreamer-capi-util.c [hunk 2]
```diff
@@ -1323,8 +1323,8 @@ ml_get_nnfw_type_by_subplugin_name (const char *name)
  * @brief Checks the availability of the given execution environments.
```
Comment (anyj0527) [inline c/src/nnstreamer-capi-util.c:1323]: minor: `... with custom option.` is omitted

Comment (jaeyun-jung) [reply c/src/nnstreamer-capi-util.c]: updated

**Review by anyj0527 — APPROVED** [review]

**Review by again4you — APPROVED** [review]
LGTM!

File: c/include/nnstreamer.h [hunk 3]
```diff
@@ -1263,6 +1263,24 @@ int ml_tensors_data_set_tensor_data (ml_tensors_data_h data, unsigned int index,
...
+/**
+ * @brief Checks the availability of the given execution environments with custom option.
+ * @details If the function returns an error, @a available may not be changed.
+ * @since_tizen 6.5
+ * @param[in] nnfw Check if the nnfw is available in the system.
+ *               Set #ML_NNFW_TYPE_ANY to skip checking nnfw.
+ * @param[in] hw Check if the hardware is available in the system.
+ *               Set #ML_NNFW_HW_ANY to skip checking hardware.
+ * @param[in] custom_option Custom option string to check framework and hardware.
+ *                      If an nnstreamer filter plugin needs to handle detailed option for hardware detection, use this parameter.
```
Comment (dongju-chae) [inline c/include/nnstreamer.h:1275]: Question: how API users know which string format should be provided.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Review by sgchoi5 — COMMENTED** [review]
things related with ACR.. LGTM

**Comment by jaeyun-jung** [comment]
need https://github.com/nnstreamer/nnstreamer/pull/3463 to fix build error.
Prepare group SR, force-merge.

**Pull request merged**